### PR TITLE
Add ceilometer-operator jobs

### PR DESCRIPTION
**Pull request by paramite**

### CODE REVIEW

**Review by jlarriba — APPROVED** [review]
I  have tested and everything works as expected.

**Comment by viroel** [comment]
@dasm please check that we need all users from OWNERS[1] file to be added to OpenShift org, or they won't be able to vote/approve jobs in openshift/release, they are 3 users that are not part of OpenShift org.

File: ci-operator/config/openstack-k8s-operators/ceilometer-operator/openstack-k8s-operators-ceilometer-operator-main.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+base_images:
```
Comment (viroel) [inline ci-operator/config/openstack-k8s-operators/ceilometer-operator/openstack-k8s-operators-ceilometer-operator-main.yaml:1]: we are not using this pre-commit image anymore, pls check comment below.

File: ci-operator/config/openstack-k8s-operators/ceilometer-operator/openstack-k8s-operators-ceilometer-operator-main.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  openstack-operator-ci-pre-commit:
+    name: tools
+    namespace: openstack-k8s-operators
+    tag: openstack-operator-ci-pre-commit
+binary_build_commands: make build
+build_root:
```
Comment (viroel) [inline ci-operator/config/openstack-k8s-operators/ceilometer-operator/openstack-k8s-operators-ceilometer-operator-main.yaml:7]: You will need to point to repo's base root image. It is easier for devs to update golang and operator-sdk versions:

https://github.com/openshift/release/blob/ba8a7c3ff6838ea69f548ada6e96412482160281/ci-operator/config/openstack-k8s-operators/manila-operator/openstack-k8s-operators-manila-operator-main.yaml#L10-L11

File: ci-operator/config/openstack-k8s-operators/ceilometer-operator/openstack-k8s-operators-ceilometer-operator-main.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    git log -1
+    mkdir -p ../operator && cp -r . ../operator
+    cd ../operator
+    go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
+    source <(setup-envtest use -p env)
+    go get golang.stackrox.io/kube-linter/cmd/kube-linter
+    GO111MODULE=on go install golang.stackrox.io/kube-linter/cmd/kube-linter
+    pre-commit run --all-files --show-diff-on-failure
+  container:
+    from: openstack-operator-ci-pre-commit
```
Comment (viroel) [inline ci-operator/config/openstack-k8s-operators/ceilometer-operator/openstack-k8s-operators-ceilometer-operator-main.yaml:34]: You will need to add new pre-commit image like:
https://github.com/openshift/release/blob/ba8a7c3ff6838ea69f548ada6e96412482160281/ci-operator/config/openstack-k8s-operators/manila-operator/openstack-k8s-operators-manila-operator-main.yaml#L10-L11

With this approach, we can build a pre-commit image with the proper version of golang, defined in the repo config[1].

[1] https://github.com/openstack-k8s-operators/ceilometer-operator/blob/main/.ci-operator.yaml

**Review by viroel — CHANGES_REQUESTED** [review]

File: ci-operator/config/openstack-k8s-operators/ceilometer-operator/openstack-k8s-operators-ceilometer-operator-main.yaml (hunk 2 above)
Comment (paramite) [reply ci-operator/config/openstack-k8s-operators/ceilometer-operator/openstack-k8s-operators-ceilometer-operator-main.yaml:7]: ack, will fix.

**Comment by paramite** [comment]
Regarding OWNERS list, I've removed those three from OWNERS here and also from ceilometer-operator with [1].

[1] https://github.com/openstack-k8s-operators/ceilometer-operator/pull/44

File: ci-operator/config/openstack-k8s-operators/ceilometer-operator/openstack-k8s-operators-ceilometer-operator-main.yaml (hunk 2 above)
Comment (viroel) [reply ci-operator/config/openstack-k8s-operators/ceilometer-operator/openstack-k8s-operators-ceilometer-operator-main.yaml:2]: I think that you missed this change. Without the correct build_root image, any job running with `from: src` may fail

**Comment by viroel** [comment]
hi @paramite - you will need to run `make jobs` and commit the changes added by ci automation scripts. Thanks

**Comment by viroel** [comment]
@paramite pre-commit job is failing[1], can you check what needs to be fixed? pls reach me if you need help with debug.

[1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/36465/rehearse-36465-pull-ci-openstack-k8s-operators-ceilometer-operator-main-precommit-check/1635741459674042368/artifacts/test/build-log.txt

**Comment by paramite** [comment]
@viroel I guess it's fine now, isn't it?

File: ci-operator/config/openstack-k8s-operators/ceilometer-operator/openstack-k8s-operators-ceilometer-operator-main.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      memory: 200Mi
+tests:
+- as: precommit-check
+  commands: |
+    export SKIP=no-commit-to-branch
+    export GOFLAGS=
+    export PRE_COMMIT_HOME=/tmp
+    git log -1
+    mkdir -p ../operator && cp -r . ../operator
+    cd ../operator
```
[the remark this answers is not in the document]
Comment (paramite) [reply ci-operator/config/openstack-k8s-operators/ceilometer-operator/openstack-k8s-operators-ceilometer-operator-main.yaml:27]: @Sandeepyadav93 done

[the remark this answers is not in the document]
Comment (Sandeepyadav93) [reply ci-operator/config/openstack-k8s-operators/ceilometer-operator/openstack-k8s-operators-ceilometer-operator-main.yaml:27]: @paramite I think we don't need installing kube-linter here as its added in Makefile and also don't need envtest setup in CI side.

Can we sync the pre-commit job to other operators, ex: https://github.com/openshift/release/pull/36752/files#diff-6d91852798d744ed40dba1d5ed0d4a55076a2e3670e7c5eff4d34945ba09b7dd

**Comment by paramite** [comment]
@Sandeepyadav93 should be ok now

**Comment by Sandeepyadav93** [comment]
@paramite this looks good, thanks for the updates. 
Just one comment before we merge this - I think we can squash https://github.com/openshift/release/pull/36465/commits in a single commit for cleaner git history.

**Pull request merged**